File: 1041-available-captures-for-rook/available-captures-for-rook.java
class Solution {
    public int numRookCaptures(char[][] board) {
        int cnt = 0;

        for (int i = 0; i < 8; i++) {
            for (int j = 0; j < 8; j++) {
                if (board[i][j] == 'R') {
                    cnt = checkAttack(board, i, j);
                }
            }
        }

        return cnt;
    }

    private int checkAttack(char[][] board, int row, int col) {
        int cnt = 0;

        boolean canAttack = false;
        for (int i = 0; i < col; i++) {
            if (board[row][i] == 'p')
                canAttack = true;
            else if (board[row][i] == 'B')
                canAttack = false;
        }

        if (canAttack)
            cnt++;

        canAttack = false;
        for (int i = 7; i > col; i--) {
            if (board[row][i] == 'p')
                canAttack = true;
            else if (board[row][i] == 'B')
                canAttack = false;
        }

        if (canAttack)
            cnt++;

        canAttack = false;
        for (int i = 0; i < row; i++) {
            if (board[i][col] == 'p')
                canAttack = true;
            else if (board[i][col] == 'B')
                canAttack = false;
        }

        if (canAttack)
            cnt++;

        canAttack = false;
        for (int i = 7; i > row; i--) {
            if (board[i][col] == 'p')
                canAttack = true;
            else if (board[i][col] == 'B')
                canAttack = false;
        }

        if (canAttack)
            cnt++;

        return cnt;
    }
}

File: 1041-available-captures-for-rook/available-captures-for-rook.js
/**
 * @param {character[][]} board
 * @return {number}
 */
var numRookCaptures = function (board) {
    let cnt = 0;
    let rookX = -1, rookY = -1;

    // Find the Rook's position
    for (let i = 0; i < 8; i++) {
        for (let j = 0; j < 8; j++) {
            if (board[i][j] === 'R') {
                rookX = i;
                rookY = j;
                break;
            }
        }
        if (rookX !== -1) break;
    }

    // Directions: Up, Right, Down, Left
    const directions = [
        [-1, 0], [0, 1], [1, 0], [0, -1]
    ];

    for (let [dx, dy] of directions) {
        let x = rookX, y = rookY;
        while (true) {
            x += dx;
            y += dy;
            if (x < 0 || x >= 8 || y < 0 || y >= 8 || board[x][y] === 'B') break;
            if (board[x][y] === 'p') {
                cnt++;
                break;
            }
        }
    }

    return cnt;
};
